src/actions/loadData.js

```diff
@@ -74,184 +75,118 @@ export const collectDatasetFetchUrls = (url) => {
-  try {
-    const availableDatasets = await fetchJSON(`${getServerAddress()}/getAvailable?prefix=${window.location.pathname}`);
-    dispatch({type: types.SET_AVAILABLE, data: availableDatasets});
-  } catch (err) {
-    console.error("Failed to fetch available datasets", err.message);
-    dispatch(warningNotification({message: "Failed to fetch available datasets"}));
+/**
+ * Trigger one or more dispatches
+ * intended to display a (non-narrative) visualisation.
+ * The dispatches may be to a 404 / splash page if the
+ * (main) dataset request fails.
+ */
+async function loadDatasetVisualisation(dispatch, url, query) {
+
+  /* step 1: parse URL into api requests to make */
+  warnDeprecatedQuerySyntax(dispatch, query);
+  const [mainDatasetUrl, secondTreeUrl] = collectDatasetFetchUrls(url);
+  const apiCalls = parseUrlIntoAPICalls(mainDatasetUrl);
+  if (secondTreeUrl) {
+    apiCalls.second = parseUrlIntoAPICalls(secondTreeUrl).main;
   }
 
-  /* Attempt to fetch the root-sequence JSON, which may or may not exist */
+  /* step 2: fetch main JSON (or mainJSONs if 2 trees) */
+  let mainJson, secondJson, pathnameShouldBe, additionalRequests;
   try {
-    const rootSequenceData = await getDataset(mainDatasetUrl, {type: "root-sequence"})
-      .then((res) => res.json());
-    dispatch({type: types.SET_ROOT_SEQUENCE, data: rootSequenceData});
-    dispatch(updateColorByWithRootSequenceData());
+    ({mainJson, secondJson, pathnameShouldBe, additionalRequests} =
+      await fetchDatasetPair(apiCalls.main, apiCalls.second));
   } catch (err) {
-    // We don't log anything as it's not unexpected to be missing the root-sequence JSON
-    // console.log("Failed to get the root-sequence JSON", err.message);
+    handleFetchErrors(err, dispatch, `Couldn't load JSONs for ${url}`);
+    return;
   }
 
-  return undefined;
-};
+  /* step 3: dispatch (viz should render) */
+  dispatchCleanStart(dispatch, mainJson, secondJson, pathnameShouldBe, query);
 
-export const loadSecondTree = (secondTreeUrl, firstTreeUrl) => async (dispatch, getState) => {
-  let secondJson;
-  try {
-    secondJson = await getDataset(secondTreeUrl)
-      .then((res) => res.json());
-  } catch (err) {
-    console.error("Failed to fetch additional tree", err.message);
-    dispatch(warningNotification({message: "Failed to fetch second tree"}));
-    return;
+  /* step 4: trigger promises to fetch sidecar files. These may resolve in dispatches to
+  update the visualisation or to show a notification */
+  if (additionalRequests.tipFrequencies) {
+    fetchJSON(apiCalls.tipFrequencies)
+      .then((data) => dispatch(loadFrequencies(data)))
+      .catch((err) => {
+        console.error("Failed to fetch frequencies", err.message);
+        dispatch(warningNotification({message: "Failed to fetch frequencies"}));
+      });
+  }
+  if (additionalRequests.rootSequence) {
+    fetchJSON(apiCalls.rootSequence)
+      .then((data) => dispatch({type: types.SET_ROOT_SEQUENCE, data}))
```

Comment:
Do we need to also dispatch `updateColorByWithRootSequenceData` here as in https://github.com/nextstrain/auspice/pull/1305/files#diff-7e6b4cc45e1472da7b3f8f6e7438315586ecbd5450bc7db2a5263a6fe2063e16L221 ?

Reply:
Yes we do! Now resolved.